### PR TITLE
Fix: podAntiAffinity conflict bug

### PR DESCRIPTION
**Pull request by zoumo**
build: add compile information && update Makefile
fix: podAntiAffinity conflict bug

### CODE REVIEW

**Comment by zoumo** [comment]
@kdada 
/cc @ddysher

File: Makefile [hunk 1]
```diff
@@ -1,12 +1,31 @@
...
-.PHONY: image
-image:
-	GOOS=linux GOARCH=amd64 go build -i -v -o $(target) github.com/caicloud/loadbalancer-controller/cmd/controller 
-	docker build -t $(target) -f Dockerfile .
-	docker tag $(target) cargo.caicloudprivatetest.com/caicloud/$(target)
-	docker push cargo.caicloudprivatetest.com/caicloud/$(target)
+
+all: push
+
+RELEASE?=0.1.0
```
Comment (kdada) [inline Makefile]: `RELEASE` can be got by `git describe --tags --abbrev=0` form tags.

File: controller.yaml [hunk 2]
```diff
@@ -18,5 +18,5 @@ spec:
             requests:
               cpu: 200m
               memory: 200Mi
-          image: cargo.caicloudprivatetest.com/caicloud/loadbalancer-controller:latest
+          image: cargo.caicloud.io/caicloud/loadbalancer-controller:0.1.0
```
Comment (kdada) [inline controller.yaml]: For convention, the version number should be `v0.1.0`

File: version/version.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package version
+
+var (
```
Comment (kdada) [inline version/version.go]: Use const.

File: provider/providers/ipvsdr/ipvsdr.go [hunk 4]
```diff
@@ -367,6 +367,18 @@ func (f *ipvsdr) cleanup(lb *netv1alpha1.LoadBalancer) error {
 		f.client.ExtensionsV1beta1().Deployments(d.Namespace).Delete(d.Name, &metav1.DeleteOptions{})
 	}
 
+	// clean up rs
+	// there is a bug in k8s, deployment can not cleanup all replicasets sometimes
```
Comment (kdada) [inline provider/providers/ipvsdr/ipvsdr.go]: Where is the issue?

File: provider/providers/ipvsdr/ipvsdr.go [hunk 5]
```diff
@@ -460,21 +474,13 @@ func (f *ipvsdr) generateDeployment(lb *netv1alpha1.LoadBalancer) *extensions.De
 							ImagePullPolicy: v1.PullAlways,
 							Resources: v1.ResourceRequirements{
 								Limits: v1.ResourceList{
-									v1.ResourceCPU:    resource.MustParse("1000m"),
-									v1.ResourceMemory: resource.MustParse("500Mi"),
+									v1.ResourceCPU:    resource.MustParse("200m"),
```
Comment (kdada) [inline provider/providers/ipvsdr/ipvsdr.go]: Fixed constant should be specified by env or cmd flags.

File: proxy/proxies/nginx/nginx.go [hunk 6]
```diff
@@ -602,24 +632,24 @@ func (f *nginx) GenerateDeployment(lb *netv1alpha1.LoadBalancer) *extensions.Dep
...
-							LivenessProbe: &v1.Probe{
-								Handler: v1.Handler{
-									HTTPGet: &v1.HTTPGetAction{
-										Path:   "/ingress-controller-healthz",
-										Port:   intstr.FromInt(10254),
-										Scheme: v1.URISchemeHTTP,
-									},
-								},
-							},
+							// ReadinessProbe: &v1.Probe{
```
Comment (kdada) [inline proxy/proxies/nginx/nginx.go]: Why do you remove health check?

Comment (zoumo) [inline proxy/proxies/nginx/nginx.go]: I have no idea why health check can not work now, and I don't have the time to track down the problem. So I remove it temporarily

Comment (kdada) [inline proxy/proxies/nginx/nginx.go]: ok

File: Makefile (hunk 1 above)
Comment (zoumo) [inline Makefile]: Unfortunately, there are no tags...

Comment (zoumo) [inline Makefile]: I will add a TODO on it

File: version/version.go (hunk 3 above)
Comment (zoumo) [inline version/version.go]: No. const is not correct.
RELEASE, REPO,COMMIT will be changed during the compilation.
See Makefile

File: provider/providers/ipvsdr/ipvsdr.go (hunk 4 above)
Comment (zoumo) [inline provider/providers/ipvsdr/ipvsdr.go]: No issue refered. I found it in practice

File: version/version.go (hunk 3 above)
Comment (kdada) [inline version/version.go]: `ld` can modify constants in compile-time.

Comment (zoumo) [inline version/version.go]: Unfortunately, it can't

**Comment by zoumo** [comment]
The following issue will be fix in next PR:
- Fixed constant should be specified by env or cmd flags

**Comment by kdada** [comment]
Ok. Merge it.

**Pull request merged**